meerk40t/core/node/elem_text.py

```diff
@@ -162,17 +195,237 @@ def font(self):
+
+        if step_y < 0:
+            # if step_y is negative, translate.
+            actualized_matrix.post_translate(0, -image_height)
+        if step_x < 0:
+            # if step_x is negative, translate.
+            actualized_matrix.post_translate(-image_width, 0)
+
+        # If crop applies, apply crop.
+        cbox = self._get_crop_box(image)
+        if cbox is not None:
+            width = cbox[2] - cbox[0]
+            height = cbox[3] - cbox[1]
+            if width != image.width or height != image.height:
+                image = image.crop(cbox)
+                # We did not crop the image so far, but we already applied
+                # the cropped transformation! That may be faulty, and needs to
+                # be corrected at a later stage, but this logic, even if clumsy
+                # is good enough: don't shift things twice!
+                if orgbox[0] == 0 and orgbox[1] == 0:
+                    actualized_matrix.post_translate(cbox[0], cbox[1])
+
+        actualized_matrix.post_scale(step_x, step_y)
+        actualized_matrix.post_translate(tx, ty)
+        # # make white transparent
+        import numpy as np
+        x = np.asarray(image.convert('RGBA')).copy()
+        x[:, :, 3] = (255 * (x[:, :, :3] != 255).any(axis=2)).astype(np.uint8)
+        image = Image.fromarray(x)
+        return actualized_matrix, image
+
+
+    def process_image(self, step_x, step_y):
+        """
+        SVG matrices are defined as follows.
+        [a c e]
+        [b d f]
+
+        Pil requires a, c, e, b, d, f accordingly.
+
+        There is a small amount of slop at the edge of converted images sometimes, so it's essential
+        to mark the image as inverted if black should be treated as empty pixels. The scaled down image
+        cannot lose the edge pixels since they could be important, but also dim may not be a multiple
+        of step level which requires an introduced empty edge pixel to be added.
+        """
+
+        from PIL import Image
+
+        try:
+            actualized_matrix, image = self._process_image(step_x, step_y)
+            inverted_main_matrix = Matrix(self.matrix).inverse()
+            self._processed_matrix = actualized_matrix * inverted_main_matrix
+            self._processed_image = image
+            self._process_image_failed = False
+            bb = self.bbox()
+            self._bounds = bb
+            self._paint_bounds = bb
+        except (
+            MemoryError,
+            Image.DecompressionBombError,
+            ValueError,
+            ZeroDivisionError,
+        ) as e:
+            # Memory error if creating requires too much memory.
+            # DecompressionBomb if over 272 megapixels.
+            # ValueError if bounds are NaN.
+            # ZeroDivide if inverting the processed matrix cannot happen because image is a line
+            # print (f"Error: {e}")
+            self._process_image_failed = True
+
+    @property
+    def active_image(self):
```

Comment:
This should maybe implement the `as_image` api for the conversion stuff. 

I'd probably add it myself at some point.